PORTAL/jobs/__main__.py

```diff
@@ -916,10 +921,15 @@ def handle_args(args, parser):
 
 
 def parse_args(
-        argv: Sequence[str] = sys.argv[1:],
-        prog: str = sys.argv[0]
+        argv: Optional[Sequence[str]] = None,
+        prog: Optional[str] = None
 ) -> Tuple[str, Dict[str, Any], str, str, int, str, bool]:
 
+    if argv is None:
+        argv = sys.argv[1:]
+    if prog is None:
+        prog = sys.argv[0]
```

Comment:
Why is this necessary?  (I'm not opposed, just wondering.)

Reply:
This refactor here is to put the `if __name__ == '__main__'` clause in a function so I can run the top level of `__main__.py` providing my own arguments from a test.  I could have put these in the kwarg defaults, but I find that can be the source of subtle bugs if `sys.argv` were mutated and that wouldn't be picked up.

Reply:
Ah, you're saying `sys.argv` might have been changed between the time the module was loaded and `if __name__ == '__main__':` was executed.  (That's the only difference from the existing code that I see.)  That's a very unlikely race when run as a script.  When we run `main()` in tests we're passing in `argv` anyway, aren't we?  So a race shouldn't matter.  I'm probably still missing something here.

Reply:
I thinking more that a test calls the top level multiple times -- but on second thought a test is unlikely to mutate `sys.argv`.  So it's probably not so relevant.  I'll remove the handling of `None`.